pkg/storage/batch.go

```diff
@@ -22,6 +24,55 @@ import (
 	"github.com/grafana/loki/pkg/logql/stats"
 )
 
+type ChunkMetrics struct {
+	refs         prometheus.Counter
+	filteredRefs prometheus.Counter
+	chunks       *prometheus.CounterVec
+	batches      prometheus.Histogram
+}
+
+const (
+	statusFiltered = "filtered"
+	statusMatched  = "matched"
+)
+
+func NewChunkMetrics(r prometheus.Registerer, maxBatchSize int) *ChunkMetrics {
+	buckets := 5
+	if maxBatchSize < buckets {
+		maxBatchSize = buckets
+	}
+
+	return &ChunkMetrics{
+		refs: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: "loki",
+			Subsystem: "store",
+			Name:      "chunk_refs_downloaded_total",
+			Help:      "Number of chunks refs downloaded.",
+		}),
+		filteredRefs: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: "loki",
+			Subsystem: "store",
+			Name:      "chunk_refs_filtered_downloaded_total",
```

Comment:
Should we call this something like `chunk_refs_post_filtering_total` because the refs are already, we are just filtering them? Just a suggestion, would love to hear your thoughts on it.

Reply:
I think on this same line of thinking `chunk_refs_pre_filtering` and `chunk_refs_post_filtering`  seem intuitive to me?

Also the "Subsystem" `store` would `index` make this easier to understand in the future?